mcstatus/protocol/connection.py

```diff
@@ -231,15 +231,22 @@ def __del__(self):
             pass
 
 
+class NoConnectionsPossibleError(Exception):
+    pass
```

Comment:
This should probably take the address and pass it as the message to exception as something like: `The address {address} doesn't point anywhere, can't make a connection.`

Reply:
That sounds reasonable, and should we move this into `mcstatus.exceptions`?

Reply:
Maybe, but we don't really have any other exceptions for now afaik, so it may not really be worth it to make that file just to hold 1 exception (maybe 2 if there'd be some base `McStatusException` class.

Reply:
I'm coming from the perspective of being a fan of custom exceptions that abstract away external third-party exceptions. Of course, preserving the stacktrace of the original exception.

Reply:
If this is the route we'd like to go down, it has to start somewhere and I'd rather not move it again later.

Reply:
I didn't think we were going to make a release right after this PR is fixed so I didn't consider temporarily having it here a problem, but yeah, that's a fair argument, if you do plan on handling other exceptions later, let's move it there then